tools/autograd/templates/VariableType.cpp

```diff
@@ -378,9 +362,8 @@ template<typename... Args> inline variable_list flatten(Args&&... args) {
   return out; // RVO
 }
 
-static void increment_version(const Tensor & t) {
-  auto& var = static_cast<const Variable&>(t);
-  var.version_counter().increment();
+static void increment_version(Tensor & t) {
+  static_cast<Variable&>(t).bump_version();
```

Comment:
While we have the patient open, can we centralize all occurrences of `static_cast<Variable&>(tensor)` into an inline function `Variable& castToVariableRef(Tensor&)` so that we can add a test that Tensor really is a Variable in DEBUG mode?